### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by JuanitoFatas**
- Remove non-existent architecture keyword
- Upgrade actions/checkout to v2
- Use [cache dependencies automatically](https://github.com/ruby/setup-ruby#caching-bundle-install-automatically) by ruby/setup-ruby
- Remove CUSTOM_RUBY_VERSION 
- No need to install Bundler

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,28 +18,16 @@ jobs:
...
-      - name: Dump CUSTOM_RUBY_VERSION
-        run: echo $CUSTOM_RUBY_VERSION
-      - name: Cache gem bundle
-        uses: actions/cache@v1
-        with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-${{ matrix.ruby }}-bundle-${{ hashFiles('Gemfile.lock') }}
-      - name: Install bundler
-        run: gem install bundler --no-document
       - name: Install gem bundle
```
Comment (yous) [inline .github/workflows/ci.yml:31]: It seems that `bundle install` will be ran automatically by `ruby/setup-ruby`: https://github.com/ruby/setup-ruby#caching-bundle-install-automatically

Comment (JuanitoFatas) [reply .github/workflows/ci.yml:31]: Hmm I think we still need to run `bundle install` given https://github.com/ruby/setup-ruby/blob/a195098f6b1e9074390215a49c1c14e82151a289/.github/workflows/test.yml#L74

Comment (yous) [reply .github/workflows/ci.yml:31]: They're also running `bundle install` twice I think. 😅

That line is added [8 months ago](https://github.com/ruby/setup-ruby/blame/a195098f6b1e9074390215a49c1c14e82151a289/.github/workflows/test.yml#L74), but `bundle-cache: true` is added [5 months ago](https://github.com/ruby/setup-ruby/blame/a195098f6b1e9074390215a49c1c14e82151a289/.github/workflows/test.yml#L42). Seeing this commit, it runs `bundle install` automatically. https://github.com/ruby/setup-ruby/commit/d57231c9a394918decfec234a36c9c79116254b7#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R56

On "Checks" tab, clicking "Setup Ruby" shows the following:
```
Run ruby/setup-ruby@v1
Downloading Ruby
Extracting Ruby
Installing Bundler
bundle install
```
Clicking "Install gem bundle" shows every entry with `Using ...`.

Comment (JuanitoFatas) [reply .github/workflows/ci.yml:31]: Oh you’re absolutely right. Thanks for detailed explanations. I will add a follow-up PR.

Comment (eregon) [reply .github/workflows/ci.yml:31]: Yes, `gem install bundler & bundle install` is not necessary with `bundler-cache: true`.
I'll update that test file or add a comment.

**Pull request merged**